sdks/python/apache_beam/io/gcp/bigquery_write_it_test.py

```diff
@@ -454,6 +454,56 @@ def test_big_query_write_insert_errors_reporting(self):
           | 'ParseErrors' >> beam.Map(lambda err: (err[1], err[2])),
           equal_to(bq_result_errors))
 
+
+  @pytest.mark.it_postcommit
+  def test_big_query_write_insert_not_found_errors(self):
+    """
+    Test that NotFound errors returned by beam.io.WriteToBigQuery
+    contain both the failed rows amd the reason for it failing.
```

Comment:
Thanks for the feedback, @ahmedabu98. I think we could raise 429, 500, and 503 errors to let the pipeline errors be retried indefinitely and route all the others to the failed rows tag. This would be consistent with [errors defined as transient in API Core Retry](https://github.com/googleapis/python-api-core/blob/main/google/api_core/retry.py#L113-L135) and [AIP-194](https://google.aip.dev/194)

Those retryable errors would all be caught in the first except clause. The second clause catching [HttpError](https://github.com/google/apitools/blob/master/apitools/base/py/exceptions.py#L111-L117) type doesn't contain anything that is typically considered transient and they'll go straight to failed rows.

Reply:
Thinking about this some more along with the suggestion to add to `_NON_TRANSIENT_ERRORS`, should be able to use the response reasons to let [RetryStrategy](https://github.com/apache/beam/blob/26b77723445cf383365098b296b9db77409af94c/sdks/python/apache_beam/io/gcp/bigquery.py#L1650-L1653) do its thing.